Softeq.XToolkit.PushNotifications/Abstract/IPushNotificationsService.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System.Threading.Tasks;
+
+namespace Softeq.XToolkit.PushNotifications.Abstract
+{
+    public interface IPushNotificationsService
+    {
+        /// <summary>
+        ///     Registers application for push notifications asynchronously.
+        /// </summary>
+        /// <returns>
+        ///     A <see cref="Task"/> representing the asynchronous operation.
+        /// </returns>
+        Task RegisterForPushNotificationsAsync();
```

Comment:
Remote registration fails should probably be handled within `IRemotePushNotificationService` implementation.
And for the local registration fails - can we really recover if system failed to register for push-notifications?

Reply:
So, if I understand correctly, you suggest to return logic of waiting until push token obtained (or not) before completing the registration on both platforms? I guess it should be fine, although Android doesn't seem to have any fail indication - we might wait endlessly if something go wrong

Reply:
Currently for Android in the case when the token is null -> uses a method from the base service to log about this: 
https://github.com/Softeq/XToolkit.WhiteLabel/blob/4f4539ba0ebe05fca6e29acfd51eda70dba11b4d/Softeq.XToolkit.PushNotifications/PushNotificationsServiceBase.cs#L101-L104
https://github.com/Softeq/XToolkit.WhiteLabel/blob/4f4539ba0ebe05fca6e29acfd51eda70dba11b4d/Softeq.XToolkit.PushNotifications/PushNotificationsServiceBase.cs#L111-L115

New implementation can't notify about it:
```cs
public async Task RegisterAsync()
{
    var token = ...

    if (token != null)
    {
        _pushNotificationsConsumer.OnPushTokenRefreshed(token.ToString());
    }
}
```

Reply:
Previous implementation just ignored the fact that token is `null` during the initial registration, the same as new one:
https://github.com/Softeq/XToolkit.WhiteLabel/blob/4f4539ba0ebe05fca6e29acfd51eda70dba11b4d/Softeq.XToolkit.PushNotifications.Droid/DroidPushNotificationsService.cs#L72-L88
And when the token is refreshed through the `FirebaseMessagingService.OnNewToken()` - it seems like in the official documentation it is explicitly marked as not being null: https://firebase.google.com/docs/cloud-messaging/android/client#monitor-token-generation
To be honest, I also didn't find any indication that during initial registration token can be null either (https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/FirebaseMessaging#getToken()), seems like if token is missing - it will be generated anew, and if there's a problem with token generation - exception will be thrown

Reply:
Thanks for the investigation.
Now I see that the old SDK method can return null.